crates/autopilot/src/run_loop.rs

```diff
@@ -318,59 +309,102 @@ impl RunLoop {
             )
             .await;
 
-        let futures = self
-            .drivers
-            .iter()
-            .enumerate()
-            .map(|(index, driver)| async move {
-                let result =
-                    match tokio::time::timeout(SOLVE_TIME_LIMIT, driver.solve(request)).await {
-                        Ok(inner) => inner,
-                        Err(_) => Err(anyhow!("timeout")),
-                    };
-                (index, result)
-            })
-            .collect::<Vec<_>>();
-        let results = futures::future::join_all(futures).await;
-        results
-            .into_iter()
-            .filter_map(|(index, result)| match result {
-                Ok(result) => Some((index, result)),
+        futures::future::join_all(self.drivers.iter().map(|driver| async move {
+            let start = Instant::now();
```

Comment:
nit: Is there a reason the current time needs to be fetched for each solver individually. I think there is no meaningful loss in precision if we were to fetch the time once before building the futures.

Reply:
d84b7dc3d95bb7aeb93b7940106a6174e71d69b6